javalin/src/main/java/io/javalin/compression/Compressor.kt

```diff
@@ -0,0 +1,14 @@
+package io.javalin.compression
+
+import java.io.OutputStream
+
+/** A compressor is used to compress an output stream */
+interface Compressor {
+
+    /**  @return the compression type for this compressor */
+    fun type(): CompressionType
```

Comment:
Does this need to be a type? I see these calls to encoding related functions:

```
ctx.header(Header.CONTENT_ENCODING, it.type().typeName)
acceptedEncoding.contains(it.type().typeName, ignoreCase = true)
```

And it would be nicer if it was just

```
ctx.header(Header.CONTENT_ENCODING, it.encoding)
acceptedEncoding.contains(it.encoding, ignoreCase = true)
```

Reply:
The class `JettyPrecompressingResourceHandler` uses `CompressionType` :
```kotlin
val outputStream: OutputStream =
    compressionStrategy.compressors.forType(type)?.compress(byteArrayOutputStream)
    ?: byteArrayOutputStream
```

```kotlin
return compressedFiles.computeIfAbsent(target + type.extension) { getCompressedByteArray(resource, type) }
```

Reply:
Sure, but it could just as well use encoding? The reason I want to move away from the enum is so that new compressor could be added solely on encoding.

Reply:
There's a registry with allowed `content-encoding` values: https://www.iana.org/assignments/http-parameters/http-parameters.xhtml#content-coding

Similarly to how we have enums for registered  HTTP response codes and MIME types we could have this enum contain all posible values. If someone were to implement a new compressor it would necessarily be one of this kinds. In any other cases it shouldn't be done at the HTTP layer, but rather in the application domain layer.